modules/exploits/linux/misc/tplink_archer_a7_c7_lan_rce.rb

```diff
@@ -0,0 +1,439 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+# PR: TODO run metasploit-framework/tools/dev/msftidy.rb tplink_archerc7_lan_rce.rb
+# PR: to see and fix the stuff they don't like
+
+require 'openssl'
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::EXE
+  include Msf::Exploit::Remote::Udp
+  include Msf::Exploit::Remote::HttpServer
+  include Msf::Exploit::Remote::HttpClient
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name'           => 'TP-Link Archer A7/C7 Unauthenticated LAN Remote Code Execution',
+        'Description'    => %q{
+        This module exploits a command injection vulnerability in the tdpServer daemon (/usr/bin/tdpServer), running on
+        the router TP-Link Archer A7/C7 (AC1750), hardware version 5, MIPS Architecture, firmware version 190726.
+        The vulnerability can only be exploited by an attacker on the LAN side of the router, but the attacker does
+        not need any authentication to abuse it. After exploitation, an attacker will be able to execute any command
+        as root, including downloading and executing a binary from another host.
+        This vulnerability was discovered and exploited at Pwn2Own Tokyo 2019 by the Flashback team (Pedro Ribeiro +
+        Radek Domanski).
+        },
+        'License'        => MSF_LICENSE,
+        'Author'         =>
+        [
+          'Pedro Ribeiro <pedrib[at]gmail.com>',             # Vulnerability discovery and Metasploit module
+          'Radek Domanski <radek.domanski[at]gmail.com> @RabbitPro'     # Vulnerability discovery and Metasploit module
+        ],
+        'References'     =>
+          [
+            [ 'URL', '<TODO>'],
```

Comment:
One URL to add is:
https://www.thezdi.com/blog/2020/4/6/exploiting-the-tp-link-archer-c7-at-pwn2own-tokyo

I'll add another soon

Reply:
Here's another to my github:
https://raw.githubusercontent.com/pedrib/PoC/master/advisories/Pwn2Own/Tokyo_2019/lao_bomb.md
@rdomanski want to add yours too?